packages/datatrak-web-server/src/utils/processSurveyResponse.ts

```diff
@@ -76,31 +89,20 @@ export const processSurveyResponse = async (
     // TODO: add in photo and file upload handling, as well as adding new entities when these question types are implemented
     switch (type) {
       // format dates to be ISO strings
-      case 'SubmissionDate':
-      case 'DateOfData':
-        surveyResponse.data_time = moment(answer as string).toISOString();
-        break;
-      // add the entity id to the response if the question is a primary entity question
-      case 'PrimaryEntity': {
-        surveyResponse.entity_id = answer as string;
+      case QuestionType.SubmissionDate:
+      case QuestionType.DateOfData: {
+        const date = new Date(answer as string);
+        surveyResponse.data_time = date.toISOString();
         break;
       }
-      case 'Entity': {
-        const config = question?.config as ConfigT;
-        if (isUpsertEntityQuestion(config)) {
-          const entityObj = await buildUpsertEntity(
-            config,
-            questionId,
-            answers,
-            countryId,
-            getEntity,
-          );
-          surveyResponse.entities_upserted.push(entityObj);
-        }
-        answersToSubmit.push(answerObject);
+
+      // add the entity id to the response if the question is a primary entity question
+      case QuestionType.PrimaryEntity: {
+        const entityId = answer?.hasOwnProperty('id') ? (answer as Entity).id : (answer as string);
```

Comment:
What is the use case where the answer is an object?

Reply:
This is when a new entity has been created. When it hasn't it will be a string, because the answer will just be the entity id